arithmetization/src/main/java/net/consensys/linea/rpc/linea/LineaSetExtraData.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package net.consensys.linea.rpc.linea;
+
+import static net.consensys.linea.extradata.LineaExtraDataException.ErrorType.FAILED_CALLING_SET_EXTRA_DATA;
+import static net.consensys.linea.extradata.LineaExtraDataException.ErrorType.INVALID_ARGUMENT;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.extern.slf4j.Slf4j;
+import net.consensys.linea.extradata.LineaExtraDataException;
+import net.consensys.linea.extradata.LineaExtraDataHandler;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.hyperledger.besu.datatypes.rpc.JsonRpcResponseType;
+import org.hyperledger.besu.datatypes.rpc.RpcMethodError;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.parameters.JsonRpcParameter;
+import org.hyperledger.besu.plugin.services.RpcEndpointService;
+import org.hyperledger.besu.plugin.services.exception.PluginRpcEndpointException;
+import org.hyperledger.besu.plugin.services.rpc.PluginRpcRequest;
+
+@Slf4j
+public class LineaSetExtraData {
+
+  private static final AtomicInteger LOG_SEQUENCE = new AtomicInteger();
```

Comment:
Are timestamps not enough for this purpose?

Reply:
thinking about your question, I just found that my solution it is not working, since there could be parallel calls to RPC methods, should not be the case for this method but in theory could happen, will rework a bit.
About your question, I am not relying only log timestamp, because with them alone is not easy to reconstruct the flow of a call, since there could be multiple in parallel, so a unique id makes correlating them easier.